emitter.go

```diff
@@ -4,9 +4,13 @@ import (
 	"bytes"
 	"io"
 	"log"
+	"sync"
 	"time"
 )
 
+var wg sync.WaitGroup
```

Comment:
Global wg is not good idea. lets move it to Start function and pass &wg to Close func